tests/pipe.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+var test = require('tape');
+
+var Buffer = require('buffer').Buffer;
+var nock = require('nock');
+var PassThrough = require('readable-stream').PassThrough;
+var request = require('request');
+
+var MockRequest = require('../index.js').Request;
+var MockResponse = require('../index.js').Response;
+
+test('nock should return response', function t(assert) {
+
+    nock('http://localhost:5555')
+        .post('/derp')
+        .reply(200, '{a');
+
+    request.post('http://localhost:5555/derp', onResponse);
+
+    function onResponse(err, res, body) {
+        assert.ifError(err, 'Unexpected error in resonse');
+        assert.strictEqual('{a', String(body));
+        assert.end();
+    }
+});
+
+test('request should pipe data', function t(assert) {
+    nock('http://localhost:5555')
+        .post('/derp', '{a')
+        .reply(200, '{b');
+
+    var chunks = [];
+    var p = new PassThrough();
+    p.on('data', function accumulate(d) {
+        chunks.push(d);
+    });
+    p.on('end', function assertTest() {
+        var data = Buffer.concat(chunks);
+        assert.strictEqual(String(data), '{b');
+        assert.end();
+    });
+
+    request.post({
+        url: 'http://localhost:5555/derp',
+        body: '{a'
+    }).pipe(p);
+});
+
+test('request should pipe through hammock req', function t(assert) {
+    nock('http://localhost:5555')
+        .post('/derp', '{a')
+        .reply(200, '{b');
+
+    var req = MockRequest({
+        url: 'http://localhost:5555/derp',
+        method: 'POST'
+    });
+
+    var chunks = [];
+    var p = new PassThrough();
+    p.on('data', function accumulate(d) {
+        chunks.push(d);
+    });
+    p.on('end', function assertTest() {
+        var data = Buffer.concat(chunks);
+        assert.strictEqual(String(data), '{b');
+        assert.end();
+    });
+
+    req.pipe(request.post('http://localhost:5555/derp'))
+        .pipe(p);
+
+    req.write('{a');
+    req.end();
+});
```

Comment:
It seems there is a better SSCCE by removing them and piping from MockRequest to PassThrough, then vrerifying on passthrough.  The request and nock modules seem irrelevant to whether MockRequest is working as expected..

Reply:
It's very subtle because request writes headers. PassThrough does not receive headers or set headers on the destination. Using passthrough would only verify the body.

Reply:
that is correct. streams are about piping data not doing http things like writing headers. Why would we expect pipe to write a header?

Reply:
Streams have multiple modes of operation (if you think about object mode, for example), but you could also ask why pipe doesn't send the full tcp payload and only the body section of the underlying tcp stream. I'm not sure I fully agree with this reasoning, but the reality is that the request module does indeed read headers from the source on request and set headers on the destination on response.

If you would rather not have tests for interop with request that's fine. I do think it's nice to have an example of a heavily used module that interacts with node req/res pairs and prove that the interaction with hammock doesn't deviate.

Reply:
The interop and integration tests are a good idea. 💯 I'd like to mark them as integration tests and have them be separate from unit tests that code against the node core interfaces.  (I admit this is a bit pedantic)

I made another branch and I'm also integrating travis ci here since it will make things better.

Reply:
as for `stream.pipe()` sometimes setting headers, I don't really think that was a good idea as it mixes the stream responsibilities with the http ones.. 

and yeah, we cannot change that so let's just work with it.